File: leetcode/python/347.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """

        mapping = {}

        for num in nums:
            if num in mapping:
                mapping[num] += 1
            else:
                mapping[num] = 1

        sorted_items = sorted(mapping.items(), key=lambda x: x[1], reverse=True)

        ans = []

        for num, _ in sorted_items[:k]:
            ans.append(num)

        return ans

File: leetcode/swift/347.swift
class Solution {
    func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
    // have a key value hashmap, where key is the number, value is the number of appearances, we then sort by the value  
      var frequency : [Int: Int] = [:]

      for num in nums {
          frequency[num, default: 0] += 1
      }

      let sortedNums = frequency.keys.sorted { frequency[$0]! > frequency[$1]! }
      return Array(sortedNums.prefix(k))
    }
}
